### PR TITLE
Japanese Translation of test_practices

### PR DESCRIPTION
**Pull request by kzhirata**
### Description
Translated Some test_practices  pages to Japanese

- content/documentation/test_practices/_index.ja.md
- content/documentation/test_practices/design_strategies.ja.md
- content/documentation/test_practices/encouraged/_index.ja.md
- content/documentation/test_practices/encouraged/fresh_browser_per_test.ja.md
- content/documentation/test_practices/encouraged/mock_external_services.ja.md
- content/documentation/test_practices/discouraged/_index.ja.md

### Types of changes

- [ ] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [ ] Code example added (and I also added the example to all translated languages)
- [x] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist

- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.

### CODE REVIEW

File: website_and_docs/content/documentation/test_practices/design_strategies.ja.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+## 概要
+時間の経過とともに、プロジェクトは多数のテストが積み上がる傾向があります。 
+テストの総数が増えると、コードベースに変更を加えることが難しくなります。
+アプリケーションが正常に機能していても、1回の"単純な"変更で多数のテストが失敗する可能性があります。 
+これらの問題が避けられない場合もありますが、問題が発生した場合は、できるだけ早く稼働を再開する必要があります。 
+次のデザインパターンと戦略は、テストの作成と保守を容易にするためにWebDriverで以前に使用されています。 
+それらもあなたにとって役に立つかもしれません。
+
+[DomainDrivenDesign](encouraged/domain_specific_language.ja.md)：アプリのエンドユーザーの言語でテストを表現します。
```
Comment (harsha509) [inline website_and_docs/content/documentation/test_practices/design_strategies.ja.md]: [DomainDrivenDesign]({{< ref "//encouraged/domain_specific_language.ja.md" >}})

File: website_and_docs/content/documentation/test_practices/design_strategies.ja.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+## 概要
+時間の経過とともに、プロジェクトは多数のテストが積み上がる傾向があります。 
+テストの総数が増えると、コードベースに変更を加えることが難しくなります。
+アプリケーションが正常に機能していても、1回の"単純な"変更で多数のテストが失敗する可能性があります。 
+これらの問題が避けられない場合もありますが、問題が発生した場合は、できるだけ早く稼働を再開する必要があります。 
+次のデザインパターンと戦略は、テストの作成と保守を容易にするためにWebDriverで以前に使用されています。 
+それらもあなたにとって役に立つかもしれません。
+
+[DomainDrivenDesign](encouraged/domain_specific_language.ja.md)：アプリのエンドユーザーの言語でテストを表現します。   
+[PageObjects](encouraged/page_object_models.ja.md)：WebアプリのUIの単純な抽象化
```
Comment (harsha509) [inline website_and_docs/content/documentation/test_practices/design_strategies.ja.md]: [PageObjects]({{< ref "//encouraged/page_object_models.ja.md" >}})

Comment (harsha509) [reply website_and_docs/content/documentation/test_practices/design_strategies.ja.md]: This should fix the broken links. I have updated it for en, zh-cn, pt-br docs!

**Review by harsha509 — CHANGES_REQUESTED** [review]

File: website_and_docs/content/documentation/test_practices/design_strategies.ja.md (hunk 2 above)
Comment (kzhirata) [reply website_and_docs/content/documentation/test_practices/design_strategies.ja.md]: @harsha509 Thank you. I've fixed these broken links.

**Review by harsha509 — APPROVED** [review]

**Comment by harsha509** [comment]
Thank you @kzhirata !

**Pull request merged**